pkg/vsphere/session/session.go

```diff
@@ -328,3 +345,49 @@ func isNotAuthenticated(err error) bool {
 	}
 	return false
 }
+
+// GetESXSession returns a general-purpose ESX session for tests.
+func GetESXSession(ctx context.Context, service string) (*Session, error) {
```

Comment:
Not a fan of having test-only helpers in packages that are used outside of tests.  Sadly we have some of that already in vsphere/datastore/datastore_util.go.  The tough part would be coming up with a decent package name ;-/